frontend/institution/allInstitutionsController.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+(function() {
+    var app = angular.module('app');
+
+    app.controller("AllInstitutionsController", function AllInstitutionsController(
+        $state, InstitutionService, AuthService, MessageService) {
+        var allInstituinsCtrl = this;
+        allInstituinsCtrl.isLoadingInstitutions = true;
+
+        allInstituinsCtrl.user = AuthService.getCurrentUser();
+
+        function loadInstitutions() {
+            InstitutionService.getInstitutions().then(function success(response) {
+                allInstituinsCtrl.institutions = response.data;
+                allInstituinsCtrl.isLoadingInstitutions = false;
+            }, function error(response) {
+                $state.go('app.user.home');
+                MessageService.showToast(response.data.msg);
+            });
+        }
+
+        loadInstitutions();
+    });
+
+    app.controller("InstitutionCardController", function InstitutionCardController(
+        $state, AuthService, InstitutionService, MessageService, ngClipboard){
+        var institutionCardCtrl = this;
+        var URL_INSTITUTION = '/institution/';
+
+        institutionCardCtrl.user = AuthService.getCurrentUser();
+
+        institutionCardCtrl.goToInstitution = function goToInstitution(institutionKey) {
+            $state.go('app.institution.timeline', {institutionKey: institutionKey});
+        };
+
+        institutionCardCtrl.generateLink = function generateLink(){
+            var currentUrl = (window.location.host);
+            var url = currentUrl + URL_INSTITUTION + institutionCardCtrl.institution.key + "/home";
+            ngClipboard.toClipboard(url);
+            MessageService.showToast("O link foi copiado");
+        };
+
+        institutionCardCtrl.showFollowButton = function showFollowButton() {
+            return institutionCardCtrl.institution && !institutionCardCtrl.isMember && 
+                   institutionCardCtrl.institution.name !== "Ministério da Saúde" &&
+                   institutionCardCtrl.institution.name !== "Departamento do Complexo Industrial e Inovação em Saúde";
+         };
+
+        institutionCardCtrl.follow = function follow(){
+            var promise = InstitutionService.follow(institutionCardCtrl.institution.key);
+            promise.then(function success(){
+                MessageService.showToast("Seguindo "+ institutionCardCtrl.institution.name);
+                var institution =  {
```

Comment:
You could move this institution creation to be returned by the promise "resolve", after "follow" success